### PR TITLE
[C-Api] type of ml-information handle

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Prevent invalid function call.
Add type of ml-information handle and macro to check the handle.

### CODE REVIEW

**Review by songgot — APPROVED** [review]
LGTM

File: c/src/ml-api-common.c [hunk 1]
```diff
@@ -1583,6 +1631,10 @@ ml_option_destroy (ml_option_h option)
         "The parameter, 'option' is NULL. It should be a valid ml_option_h, which should be created by ml_option_create().");
   }
 
+  if (!ml_information_handle_is_valid (option, ML_INFO_TYPE_OPTION))
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'option' is not an ml-option handle.");
```
[the remark this answers is not in the document]
Comment (jaeyun-jung) [reply c/src/ml-api-common.c]: updated

**Comment by myungjoo** [comment]
Final check before finalizing the API formats: please check if these API names are intuitive to actual machine learning engineers.

File: c/src/ml-api-common.c [hunk 2]
```diff
@@ -19,6 +19,52 @@
...
+{
+  ml_info_type_e type; /**< The type of ml_info. */
+  unsigned int length; /**< The length of data. */
+  ml_info_s **info; /**< array of ml_info. */
+} ml_info_list_s;
+
+/**
+ * @brief Macro to check the handle has valid type of ml-information.
+ */
+#define ml_information_handle_is_valid(h,t) ((h) && *((ml_info_type_e *) (h)) == (t))
```
Comment (wooksong) [inline c/src/ml-api-common.c:66]: How about using ```static inline```?

**Review by wooksong — APPROVED** [review]
LGTM

**Pull request merged**